sql/core/src/main/scala/org/apache/spark/sql/internal/SessionState.scala

```diff
@@ -134,17 +131,20 @@ private[sql] class SessionState(
 
     SessionState.mergeSparkConf(confCopy, sparkContext.getConf)
 
-    new SessionState(
-      sparkContext,
-      newSparkSession.sharedState,
-      confCopy,
-      experimentalMethods.clone(),
-      functionRegistryCopy,
-      catalogCopy,
-      sqlParser,
-      SessionState.createAnalyzer(newSparkSession, catalogCopy, confCopy),
-      new StreamingQueryManager(newSparkSession),
-      queryExecutionCreator)
+    val newSessionState = new SessionState(
+        sparkContext,
+        newSparkSession.sharedState,
+        confCopy,
+        experimentalMethods.clone(),
+        functionRegistryCopy,
+        catalogCopy,
+        sqlParser,
+        SessionState.createAnalyzer(newSparkSession, catalogCopy, confCopy),
+        new StreamingQueryManager(newSparkSession),
+        queryExecutionCreator) {
+      override val listenerManager: ExecutionListenerManager = self.listenerManager.clone()
```

Comment:
Builders were added in [https://github.com/apache/spark/pull/17433](url). Changed initialization to use that pattern, so it is a constructor param now.